stdlib/Random/src/Xoshiro.jl

```diff
@@ -117,65 +117,30 @@ end
 
 # Shared implementation between Xoshiro and TaskLocalRNG -- seeding
 
-function seed!(x::Union{TaskLocalRNG,Xoshiro})
+function seed!(rng::Union{TaskLocalRNG,Xoshiro})
     # as we get good randomness from RandomDevice, we can skip hashing
     parent = RandomDevice()
     # Constants have nothing up their sleeve, see task.c
     # 0x02011ce34bce797f == hash(UInt(1))|0x01
     # 0x5a94851fb48a6e05 == hash(UInt(2))|0x01
     # 0x3688cf5d48899fa7 == hash(UInt(3))|0x01
     # 0x867b4bb4c42e5661 == hash(UInt(4))|0x01
-    setstate!(x,
+    setstate!(rng,
               0x02011ce34bce797f * rand(parent, UInt64),
```

Comment:
I'm unclear on why we need to do these multiplications still when the input is `RandomDevice()`. I mean clearly harmless but if the random values are truly random these values aren't any more random. This was introduced in edea2de0232841b566dbc159a07416e4aa6c5c53 so maybe @chethega can clarify? I get why they're used in `task.c` but here `parent` is not another xoshiro RNG, it's a genuinely random device. Seems fine to me to omit the multiplications here. I'll note that this can also be changed at any time since this use case is explicitly not reproducible (no seed value is given).

Reply:
Agreed, no reason for the multiplications here.